File: src/main/java/co/edu/javeriana/dw/proyecto/model/Planet.java
package co.edu.javeriana.dw.proyecto.model;

import jakarta.persistence.*;
import lombok.*;

import java.util.List;

@Entity
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@EqualsAndHashCode(of = "id")
public class Planet {
    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    private Long id;
    private String name;

    @ManyToMany
    @JoinTable(
            name = "Space_Hangar",
            joinColumns = @JoinColumn(name = "planet_id"),
            inverseJoinColumns = @JoinColumn(name = "spacecraft_id")
    )
    private List<Spacecraft> spacecrafts;

    @ManyToOne
    private Star star;

   @ManyToMany
   @JoinTable(
           name = "Market",
           joinColumns = @JoinColumn(name = "planet_id"),
           inverseJoinColumns = @JoinColumn(name = "product_id")
   )
   private List<Product> products;

   @OneToMany(mappedBy = "planet")
    private List<Market> markets;
}

File: Proyecto-Desarrollo-Web-Client/src/app/model/planet.ts
import { Market } from "./market";
import { Product } from "./product";
import { Spacecraft } from "./spacecraft";
import { Star } from "./star";

export class Planet {
  constructor(
    public id: number,
    public name: string,
    public spacecrafts: Spacecraft[],
    public star: Star,
    public products: Product[],
    public markets: Market[],
    public position: number,
    public ring: boolean,
    public size: number,
    public texture: number,
    public character: number,
    public animation: number
  ){}
}
